src/central-server/admin-service/infra-jpa/src/main/java/org/niis/xroad/cs/admin/jpa/config/CurrentHANodeNameSettingTransactionManager.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License
+ * <p>
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p>
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * <p>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.niis.xroad.cs.admin.jpa.config;
+
+import org.niis.xroad.cs.admin.api.dto.HAConfigStatus;
+import org.springframework.jdbc.datasource.JdbcTransactionObjectSupport;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.CannotCreateTransactionException;
+import org.springframework.transaction.TransactionDefinition;
+
+import javax.persistence.EntityManagerFactory;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Sets xroad.current_ha_node_name setting for each transaction.
+ * Required for history table stored procedures.
+ */
+@Component("currentHANodeNameSettingTransactionManager")
+public class CurrentHANodeNameSettingTransactionManager extends JpaTransactionManager {
```

Comment:
I would have used `spring.datasource.hikari.connection-init-sql` property (would have to create custom DataSource bean to parametrize it), but I guess this approach would work too.

Could you add a (int-)test as well to make sure it is applied?